odoo/models.py

```diff
@@ -5726,6 +5726,14 @@ def exists(self) -> Self:
         new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
         if not ids:
             return self
+
+        # Special version without query
```

Comment:
More useful comment
```suggestion
        # Cache is invalidated for unlinked records, if all records are in cache, they exist
```

Reply:
I would suggest: "if a record has data in cache, then it exists (thanks to cache consistency)"